src/open_pulse/pulse_utils.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef PULSE_UTILS_H
+#define PULSE_UTILS_H
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include "types.hpp"
+
+namespace py = pybind11;
+
+py::object expect_psi_csr(py::array_t<complex_t> data,
```

Comment:
Can we change the args to be `const &` ? Or is it some constraint from pybind?

Reply:
It is the way they do it in the pybind11 documentation. These are light wrappers created on the fly 
 when the python object is passed to the function so they do not involve an extra copy (I'll recheck just in case).

Reply:
Did you have the chance to check this?

Reply:
Yes. I checked and no extra copies were involved. In all pybind11 documentation they use this approach so I sticked with that. I cannot see a downside of using the `const &` though.